### ISSUE TITLE
prototype remembered after reduction?

### ISSUE THREAD

**Issue by justin2004** [phantomics/april#203]
```
      ⍴+⌿(⍳2)∘.× ⍬ ∘.× (⍳4)
0 4
```

```
APRIL> (april "⍴+⌿(⍳2)∘.× ⍬ ∘.× (⍳4)")
#*00
```

**Comment by phantomics** [comment]
This is certainly doable, but can you think of any other operations on an array where its prototype should be carried through? I should account for all of those cases.

**Comment by justin2004** [comment]
is it easier to not think about the prototype and instead make the 2nd expression evaluate to `(SIMPLE-ARRAY T (0 4))`?
```
APRIL> (type-of (april "(⍳2)∘.× ⍬ ∘.× (⍳4)"))
(SIMPLE-ARRAY T (2 0 4))
APRIL> (type-of (april "+⌿(⍳2)∘.× ⍬ ∘.× (⍳4)"))
(SIMPLE-ARRAY T (0 0))
```

or does that need need prototype carrying?

**Comment by phantomics** [comment]
This is fixed along with #204. It turns out the issue doesn't relate to prototypes but to the function's identity value.
